nemo/collections/nlp/models/token_classification/ner_model.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, Optional
+
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+from torch.utils.data import DataLoader
+
+from nemo.collections.common.losses import CrossEntropyLoss
+from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
+from nemo.collections.nlp.data.token_classification.token_classification_dataset import BertTokenClassificationDataset
+from nemo.collections.nlp.data.token_classification.token_classification_descriptor import TokenClassificationDataDesc
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
+from nemo.collections.nlp.modules.common import TokenClassifier
+from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
+from nemo.core.classes import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.core.neural_types import NeuralType
+from nemo.utils.decorators import experimental
+
+__all__ = ['NERModel']
+
+
+@experimental
+class NERModel(ModelPT):
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.bert_model.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.classifier.output_types
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """
+        Initializes BERT Named Entity Recognition model.
+        """
+
+        self.data_desc = TokenClassificationDataDesc(
+            data_dir=cfg.data_dir, modes=["train", "test", "dev"], pad_label=cfg.train_ds.pad_label
+        )
+        self.data_dir = cfg.data_dir
+
+        # TODO add support for sentence_piece tokenizer
+        if cfg.language_model.tokenizer == 'nemobert':
```

Comment:
Don't we have support for switching between tokenizers in the new NeMo?

Reply:
I'm going to port it in a separate PR